Kill some more unused code in function_wrapper.py

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32657 Get rid of some unused THGenerate*Type defines.
* **#32600 Kill some more unused code in function_wrapper.py**
* #32599 Kill THIntegerTensor, THDenseTensor, THDenseIndexTensor.

Differential Revision: [D19565654](https://our.internmc.facebook.com/intern/diff/D19565654)

gchanan committed:
Kill some more unused code in function_wrapper.py

[ghstack-poisoned]

gchanan committed:
Update on "Kill some more unused code in function_wrapper.py"


Differential Revision: [D19565654](https://our.internmc.facebook.com/intern/diff/D19565654)

[ghstack-poisoned]

gchanan committed:
Update on "Kill some more unused code in function_wrapper.py"


Differential Revision: [D19565654](https://our.internmc.facebook.com/intern/diff/D19565654)

[ghstack-poisoned]

gchanan committed:
Update on "Kill some more unused code in function_wrapper.py"


Differential Revision: [D19565654](https://our.internmc.facebook.com/intern/diff/D19565654)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit e921b917:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e921b917de8addcc9a0f50fcb44d02e15e310997).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4347600) pytorch_linux_xenial_py3_clang5_asan_build (1/1)
**Step:** "Set Up CI Environment After attach_workspace" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=91328077) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=91328077)) :snowflake:
<details>
<summary>
<code>E: Failed to fetch https://nvidia.github.io/nvidia-container-runtime/ubuntu16.04/amd64/./nvidia-container-runtime_2.0.0+docker18.09.4-1_amd64.deb  GnuTLS recv error (-54): Error in the pull function.</code>
</summary>

```
                                                                   91% [Working]               Get:26 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 intel-microcode amd64 3.20191115.1ubuntu0.16.04.2 [2,408 kB] 
                                             94% [Working]               Get:27 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 amd64-microcode amd64 3.20191021.1+really3.20180524.1~ubuntu0.16.04.2 [30.8 kB] 
94% [27 amd64-microcode 23.5 kB/30.8 kB 76%]                       21.1 MB/s 0s 94% [Connecting to archive.ubuntu.com (91.189.88.162)]             21.1 MB/s 0s                                                                                 Get:28 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 linux-image-generic amd64 4.4.0.173.181 [2,486 B] 
94% [28 linux-image-generic 2,486 B/2,486 B 100%]                  21.1 MB/s 0s 95% [Working]                                                      21.1 MB/s 0s                                                                                 Get:29 http://archive.ubuntu.com/ubuntu xenial/universe amd64 moreutils amd64 0.57-1 [55.0 kB] 
95% [29 moreutils 12.6 kB/55.0 kB 23%]                             21.1 MB/s 0s 95% [Working]                                                      21.1 MB/s 0s                                                                                 Get:30 http://archive.ubuntu.com/ubuntu xenial/main amd64 tcl8.6 amd64 8.6.5+dfsg-2 [14.2 kB] 
95% [30 tcl8.6 4,871 B/14.2 kB 34%]                                21.1 MB/s 0s 96% [Working]                                                      21.1 MB/s 0s                                                                                 Get:31 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 thermald amd64 1.5-2ubuntu4 [187 kB] 
96% [31 thermald 4,869 B/187 kB 3%]                                21.1 MB/s 0s 97% [Working]                                                      21.1 MB/s 0s                                                                                 Get:32 http://archive.ubuntu.com/ubuntu xenial/main amd64 tk8.6 amd64 8.6.5-1 [12.2 kB] 
97% [32 tk8.6 7,447 B/12.2 kB 61%]                                 21.1 MB/s 0s 97% [Working]                                                      21.1 MB/s 0s                                                                                 Get:33 http://archive.ubuntu.com/ubuntu xenial/main amd64 xterm amd64 322-1ubuntu1 [607 kB] 
97% [33 xterm 7,445 B/607 kB 1%]                                   21.1 MB/s 0s 98% [Working]                                                      21.1 MB/s 0s                                                                                 Fetched 129 MB in 7s (17.9 MB/s) 
W: --force-yes is deprecated, use one of the options starting with --allow instead. 
E: Failed to fetch https://nvidia.github.io/nvidia-container-runtime/ubuntu16.04/amd64/./nvidia-container-runtime_2.0.0+docker18.09.4-1_amd64.deb  GnuTLS recv error (-54): Error in the pull function. 
 
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing? 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

gchanan committed:
Update on "Kill some more unused code in function_wrapper.py"


Differential Revision: [D19565654](https://our.internmc.facebook.com/intern/diff/D19565654)

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/nn_parse.py:
```diff
@@ -287,21 +287,6 @@ def backward_declaration(base, thnn_functions, backend_types):
-    if 'im2col' in base['name']:
-        # Add input_size as parameter to im2col backwards function
-        input_size_arg = {'type': 'IntArrayRef', 'name': 'input_size', 'size': 2}
-        arguments.insert(2, input_size_arg)
```
looks like `input_size` is a declared argument of these, is that why these are unneeded? but also, how did this even work if it unconditionally shoved in a duplicate arg?

gchanan replied:
it's not needed because im2col isn't in nn.yaml anymore -- I think it was removed here: https://github.com/pytorch/pytorch/pull/21769

bhosmer commented on aten/src/ATen/function_wrapper.py:
```diff
@@ -559,8 +556,6 @@ def __getitem__(self, x):
     'type_method_formals': List[str],
     'variants': str,
-    'when_spares_dispatch': str,
-    'when_sparse_dispatch': str,
```
thinking face emoji

bhosmer reviewed (approved):
Couple questions inline, but LGTM!

gchanan commented on aten/src/ATen/function_wrapper.py:
```diff
@@ -1348,17 +1339,10 @@ def get_argument(env, argument, option):
     def get_arguments(env, arguments, option):
         # type: (Environment, List[THFormal], FunctionOption) -> List[str]
         return [get_argument(env, argument, option)
-                for argument in arguments if not drop_argument(argument, option)]
```
yes, it's unused.
